### PR TITLE
build(ios): set os minimum deployment to 15.0

### PR DESCRIPTION
**Pull request by MuckT**
### Description

- Sets the min deployment target for iOS to 15.0: [Slack thread](https://valora-app.slack.com/archives/C025Q9QG8JG/p1707945581586849).
- Sets E2E tests on merges to main only run on iOS 15.0.

### Test plan

- CI E2E tests

### Related issues

- Fixes ACT-1099

### Backwards compatibility

Yes, except for iOS versions < 15.0

### CODE REVIEW

File: .github/workflows/e2e-main.yml
Comment (jeanregisser) [inline .github/workflows/e2e-main.yml:46]: Can we add iOS 17 then? Since we want to test the lowest and highest supported.

**Review by jeanregisser — APPROVED** [review]
Awesome! 🚀 

Before we merge, did you get a confirmation from product?

File: .github/workflows/e2e-main.yml
Comment (MuckT) [reply .github/workflows/e2e-main.yml:46]: This needs to be done in a follow up PR. Once we have product approval we can setup the new runners, migrate off the old ones and then enable testing of iOS 17.

**Pull request merged**
